### ISSUE TITLE
`relay-page-info` doesn't follow actual Connection specification

### ISSUE THREAD

**Issue by jukben** [B2o5T/graphql-eslint#1009]
**Describe the bug**

Apparently the [Connection](https://github.com/facebook/relay/blob/main/website/spec/Connections.md#fields-2) spec changed and `startCursor` and `endCursor` is might be nullable. 

Unfortunately, I don't know why this information is not present in the rendered version as well. (https://relay.dev/graphql/connections.htm) but I [reached out](https://discord.com/channels/625400653321076807/862957601670430731/958986159097282590) to the community to resolve this confusion. 

**To Reproduce**
Define Relay's PageInfo

```graphql
"""
Information about pagination in a connection.
"""
type PageInfo {
  """
  When paginating forwards, the cursor to continue.
  """
  endCursor: String

  """
  When paginating forwards, are there more items?
  """
  hasNextPage: Boolean!

  """
  When paginating backwards, are there more items?
  """
  hasPreviousPage: Boolean!

  """
  When paginating backwards, the cursor to continue.
  """
  startCursor: String
}
```

and then observe the issue

```
[@graphql-eslint/relay-page-info] Field `startCursor` must return non-null String or Scalar.
[@graphql-eslint/relay-page-info] Field `endCursor` must return non-null String or Scalar.
```

**Expected behavior**

Should pass as valid use-case

**Environment:**

- OS: M1
- `@graphql-eslint/eslint-plugin": "3.10.0`:
- NodeJS: v16.13.0

**Additional context**

https://github.com/rmosolgo/graphql-ruby/issues/2872#issuecomment-625812125

**Comment by jukben** [comment]
I propose to create PR losing the rule for `startCursor` and `endCursor`. I'm willing to do the PR once we agree this is something we want to do (I truly belive it is) 🚀

**Comment by B2o5T** [comment]
Thank you for reporting it. While developing this rule I was looking at this page https://relay.dev/graphql/connections.htm#sec-undefined.PageInfo.Fields

Where was it written:
> It must also contain fields startCursor and endCursor, both of which return non-null opaque strings.

Btw, I'll fix it today, feel free to share your experience with the new rules. Thank you @jukben!

**Comment by B2o5T** [comment]
@jukben @Balvajs fixed in [@graphql-eslint/eslint-plugin@3.10.1](https://github.com/B2o5T/graphql-eslint/releases/tag/%40graphql-eslint%2Feslint-plugin%403.10.1)
